pkg/kubelet/dockershim/exec.go

```diff
@@ -128,6 +130,13 @@ func (*NsenterExecHandler) ExecInContainer(client libdocker.Interface, container
 	if exitErr, ok := cmdErr.(*exec.ExitError); ok {
 		return &utilexec.ExitErrorWrapper{ExitError: exitErr}
 	}
+
+	// since native methods of command do not provide timeout mechanisms
+	// it is considered safe to wait until command ends and have a judgement of timeout after that
+	if timeout >= 0 && time.Now().Unix()-start >= int64(timeout.Seconds()) {
```

Comment:
The timeout also doesn't take effect in this change, and the change also makes actual cmdErr lost. what benefits do we get from this change?

Reply:
Well, I think the key point of the issue is that user passed in a timeout param and expected probe to fail if timeout, and then rebuild the container. The change is mainly for fix this. 
If we want timeout to take effect in command execution itself, I'm afraid we have to intrusively modify native exec func